source/common/network/address_impl.cc

```diff
@@ -175,8 +175,15 @@ std::string Ipv4Instance::sockaddrToString(const sockaddr_in& addr) {
 absl::uint128 Ipv6Instance::Ipv6Helper::address() const {
   absl::uint128 result{0};
   static_assert(sizeof(absl::uint128) == 16, "The size of asbl::uint128 is not 16.");
-  memcpy(static_cast<void*>(&result), static_cast<const void*>(&address_.sin6_addr.s6_addr),
+#ifdef __SIZEOF_INT128__
+  unsigned __int128 ipv6_addr;
+  SAFE_MEMCPY(&ipv6_addr, &(address_.sin6_addr.s6_addr));
+  result = ipv6_addr; // Use assignment operator which defaults to copy-constructor
+#else
+  memcpy(static_cast<void*>(&result),
+         static_cast<const void*>(&address_.sin6_addr.s6_addr), // NOLINT(safe-memcpy)
          sizeof(absl::uint128));
+#endif
```

Comment:
I hadn't noticed this mess before. Can you see if this works without the ifdef?
```
SAFE_MEMCPY(reinterpret_cast<absl::uint128*>(&result),
            reinterpret_cast<absl::uint128*>(&address_.sin6_addr.s6_addr));
```
or even better, what happens if you just have:
```
SAFE_MEMCPY(&result, &address_.sin6_addr.s6_addr);
```
Would that fail in one of our CI builds? And if so, with what error message?

Reply:
The error was: 
```
copying an object of type 'class absl::uint128' with 'private' member 'absl::uint128::lo_' from an array of 'const uint8_t [16]' {aka 'const unsigned char [16]'}; use assignment or copy-initialization instead [-Werror=class-memaccess]
```  
[link to the failed pipeline](https://dev.azure.com/cncf/envoy/_build/results?buildId=57183&view=logs&j=8bf29878-a4cc-50f7-4e84-2255e6fd4065&t=3267cd16-a387-57e0-0ef4-d7a4d1c6c089&l=8272).

Reply:
that's what happens as the result of calling SAFE_MEMCPY as suggested above?

I might have expected that if you just try to do an assign.